test/simple/test-tls-cipher-list.js

```diff
@@ -60,11 +45,185 @@ function doTest(checklist, env, useswitch) {
       });
 }
 
+// test that the command line switchs takes precedence
+// over the environment variables
+function doTestPrecedence() {
+  // test that --cipher-list takes precedence over NODE_CIPHER_LIST
+  doTest('ABC', ['--cipher-list=ABC'], {'NODE_CIPHER_LIST': 'XYZ'});
+
+  // test that --enable-legacy-cipher-list takes precedence
+  // over NODE_CIPHER_LIST
+  doTest(V1038Ciphers,
+         ['--enable-legacy-cipher-list=v0.10.38'],
+         {'NODE_CIPHER_LIST': 'XYZ'});
+
+  // test that --cipher-list takes precedence over NODE_LEGACY_CIPHER_LIST
+  doTest('ABC',
+         ['--cipher-list=ABC'],
+         {'NODE_LEGACY_CIPHER_LIST': 'v0.10.38'});
+
+
+  // test that --enable-legacy-cipher-list takes precence over both envars
+  doTest(V1038Ciphers,
+         ['--enable-legacy-cipher-list=v0.10.38'],
+         {
+           'NODE_LEGACY_CIPHER_LIST': 'v0.10.38',
```

Comment:
In v0.10.39 there is only one legal value for this. Passing in anything but
v0.10.38 throws, even if it's overridden by the command line switch. For
v0.12.3 and master I specify different values.
On May 4, 2015 5:19 PM, "Julien Gilli" <notifications@github.com> wrote:

> In test/simple/test-tls-cipher-list.js
> <https://github.com/joyent/node/pull/14572#discussion_r29636669>:
>
> > +  // over NODE_CIPHER_LIST
> > +  doTest(V1038Ciphers,
> > +         ['--enable-legacy-cipher-list=v0.10.38'],
> > +         {'NODE_CIPHER_LIST': 'XYZ'});
> > +
> > +  // test that --cipher-list takes precedence over NODE_LEGACY_CIPHER_LIST
> > +  doTest('ABC',
> > +         ['--cipher-list=ABC'],
> > +         {'NODE_LEGACY_CIPHER_LIST': 'v0.10.38'});
> > +
> > +
> > +  // test that --enable-legacy-cipher-list takes precence over both envars
> > +  doTest(V1038Ciphers,
> > +         ['--enable-legacy-cipher-list=v0.10.38'],
> > +         {
> > +           'NODE_LEGACY_CIPHER_LIST': 'v0.10.38',
>
> Don't we want another value for NODE_LEGACY_CIPHER_LIST (e.g ABC),
> otherwise how do we know that the value for legacy cipher list was taken
> from the command line switch and not from the environment variable?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joyent/node/pull/14572/files#r29636669>.
>